dao/src/test/java/se/fortnox/reactivewizard/db/DbProxyTest.java

```diff
@@ -410,57 +478,48 @@ public void testAllPathsOfTryWithResourceUsingCount() throws SQLException {
         insertAndAssertError(insertCount(), "prepare statement");
     }
 
-    private Observable<Integer> insertCount() {
+    private Flux<Integer> insertCount() {
         return dbProxyTestDao.insertLocalDate(LocalDate.MAX);
     }
 
     private void insertAndAssertError(String message) {
         insertAndAssertError(dbProxyTestDao.insert(), message);
     }
 
-    private void insertAndAssertError(Observable<?> insert, String message) {
-        try {
-            insert.toBlocking().singleOrDefault(null);
-            fail("expected exception");
-        } catch (Exception e) {
-            Exception cause = (Exception) e.getCause();
-            if (cause != null) {
-                e = cause;
-            }
-            if (message != null) {
-                assertThat(e.getMessage()).contains(message);
-            }
-        }
+    private void insertMonoAndAssertError(String message) {
+        insertAndAssertError(dbProxyTestDao.insertMono().flux(), message);
     }
 
     private void insertAndAssertError(Flux<?> insert, String message) {
         assertThatExceptionOfType(Exception.class)
                 .isThrownBy(insert::blockLast)
                 .satisfies(exception -> {
                     Throwable selfOrCause = Optional.ofNullable(exception.getCause()).orElse(exception);
-                    assertThat(selfOrCause.getMessage()).contains(message);
+                    if (Objects.nonNull(message)) {
+                        assertThat(selfOrCause.getMessage()).contains(message);
```

Comment:
assertj goodness: hasMessageContaining :D 
Why do we want to send null message btw?

Reply:
There are some tests that want to skip the exception message assertion, and they do so by setting it to null, see [https://github.com/FortnoxAB/reactive-wizard/blob/master/dao/src/test/java/se/fortnox/reactivewizard/db/DbProxyTest.java#L430](DbProxyTest:430). Not sure why though.